core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -192,7 +192,12 @@ class BaseColumnFormatRelation(
       truncate()
     }
     JdbcExtendedUtils.saveTable(data, table, schema, connProperties)
-    flushRowBuffer()
+
+    SnappyContext.getClusterMode(_context.sparkContext) match {
+      case SnappyEmbeddedMode(_, _) => flushRowBuffer()
+      case LocalMode(_, _) => flushRowBuffer()
+      case _ =>
+    }
```

Comment:
Strange, the same call has been there in ColumnFormatSamplingRelation for very long but similar issue never reported for sample. Due to that other call I would suggest that this check go inside flushRowBuffer() itself.